### ISSUE TITLE
knowledge_distillation teacher model 

### ISSUE THREAD

**Issue by rbhambriiit** [keras-team/keras-io#755]
The teacher model trained at 
https://github.com/keras-team/keras-io/blob/master/examples/vision/knowledge_distillation.py
https://github.com/keras-team/keras-io/blob/master/examples/vision/knowledge_distillation.py#L166

does not have a softmax layer at the Dense layer with 10 possible classes.
Is that expected?

**Comment by rbhambriiit** [comment]
is this internally covered here:

![image](https://user-images.githubusercontent.com/13808100/148030634-541bc1ce-36f4-4299-9d19-a28ca09d3d8e.png)

**Comment by sachinprasadhs** [comment]
Hi, 

Thanks for reporting the issue.

Yes, as you have pointed out, it is handled at `loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True)`.

Initially, the model is just created without any activation function. Which results in linear output.

```
teacher = keras.Sequential(
    [
        keras.Input(shape=(28, 28, 1)),
        layers.Conv2D(256, (3, 3), strides=(2, 2), padding="same"),
        layers.LeakyReLU(alpha=0.2),
        layers.MaxPooling2D(pool_size=(2, 2), strides=(1, 1), padding="same"),
        layers.Conv2D(512, (3, 3), strides=(2, 2), padding="same"),
        layers.Flatten(),
        layers.Dense(10),
    ],
    name="teacher",
)
```

So when the model is compiled, it is required to use from_logits = True, to define the inputs are linear.

```
# Train teacher as usual
teacher.compile(
    optimizer=keras.optimizers.Adam(),
    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
    metrics=[keras.metrics.SparseCategoricalAccuracy()],
)
```
